### PR TITLE
Position features

### PR DESCRIPTION
**Pull request by dawahs**
Ok, here's the big one: position-based features. Most files have only minor changes based on the input to `_get_feature_dict_for_region` needing an additional variable (it needs the "raw position" from before processing the sequence).

I modified the unit tests for the residuals, and everything looks good. There was a minor change in the processing of degenerate nucleotides so the position information wasn't lost after processing, and so that unit test also needed changing.

Really the major change is `position_feature_generator.py`.

### CODE REVIEW

File: position_feature_generator.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            label_dict[lab].append((start, end))
+
+        all_feature_info_list = [(lab, break_list) for lab, break_list in sorted(label_dict.iteritems())]
+        self.feature_info_list = [feat_tuple for feat_tuple in all_feature_info_list if feat_tuple not in feats_to_remove]
+
+        self.pos_dict = {i: None for i in range(self.max_seq_len)}
+        for idx, (lab, break_list) in enumerate(self.feature_info_list):
+            for pos in range(self.max_seq_len):
+                for start, end in break_list:
+                    if pos in range(start, end):
```
Comment (jjfeng) [inline position_feature_generator.py]: `if pos >= start and pos< end`?

File: position_feature_generator.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        """
+        label_dict = {lab: [] for lab in self.labels}
+        for start, end, lab in izip(self.breaks, self.breaks[1:], self.labels):
+            label_dict[lab].append((start, end))
+
+        all_feature_info_list = [(lab, break_list) for lab, break_list in sorted(label_dict.iteritems())]
+        self.feature_info_list = [feat_tuple for feat_tuple in all_feature_info_list if feat_tuple not in feats_to_remove]
+
+        self.pos_dict = {i: None for i in range(self.max_seq_len)}
+        for idx, (lab, break_list) in enumerate(self.feature_info_list):
```
Comment (jjfeng) [inline position_feature_generator.py]: this can be made a lot faster i think but i dont really know if it matters.

File: models.py [hunk 3]
```diff
@@ -83,6 +81,17 @@ def __init__(self, start_seq, end_seq, motif_len=3, left_flank_len=None, right_f
         self.end_seq_with_flanks = self.left_flank + end_seq + self.right_flank
         self.seq_len = len(self.start_seq)
         self.collapse_list = collapse_list
+        self.raw_pos = {}
```
Comment (jjfeng) [inline models.py]: maybe add more comments for what `raw_pos` is

File: position_feature_generator.py (hunk 2 above)
Comment (dawahs) [reply position_feature_generator.py]: Yah, changed. It's only run once really but for extra-long sequences this change is way faster.

**Pull request merged**